multicluster/controllers/multicluster/commonarea/acnp_resourceimport_controller.go

```diff
@@ -43,6 +43,10 @@ var (
 )
 
 func (r *ResourceImportReconciler) handleResImpUpdateForClusterNetworkPolicy(ctx context.Context, resImp *multiclusterv1alpha1.ResourceImport) (ctrl.Result, error) {
+	if resImp.Spec.ClusterNetworkPolicy == nil {
```

Comment:
Questions:

1. Do we need this check for ResourceExport reconcilers too?
2. But what will happen if this ResourceImport was not processed before mc-controller upgrade? After such change, we should either recreate all impacted resources or support old version?

Reply:
We can skip reconciling ResourceExport if spec is empty, but the main purpose of this change is to avoid controller crash for any existing ResourceImport with empty spec. When users deploy latest manifest with this change, the existing ResourceImport spec will become empty due to mismatch json tag.
When users upgrade all member and leader clusters, they need to restart controller manually, then all impacted resources will be updated automatically because ResourceExports will be updated by member controllers.

Reply:
I am thinking maybe we can add an empty line in the config file so that the ConfigMap checksum annotation can be updated. In this case, user don't have to restart controller manually. Deployment will be updated and restarted automatically due to annotation change. What's your thought?

Reply:
After upgrading from 1.8.0 to 1.9.0 which includes this change, mc-controller will be restarted. Why manual restart is required? But could you confirm in the current implementation, mc-controller will update all existing ResourceExports after restart? Could you verify that?

Reply:
Oh, yes, you are right. For this standalone PR, the manifest change is for CRD part only, so controller won't be restarted. But for a new release 1.9, mc-controller will be restarted.
I verified this locally, when the leader CRD is updated, the spec of existing ResourceExport will be empty, but it will be updated soon after member controller restart since there are difference between new ResourceExport and existing one.

Reply:
So, will ResourceImport be recreated too? How about imported resources in the member cluster?

Reply:
The corresponding ResourceImport will be updated, then the member controller will be notified and handles the update event properly to refresh imported resources.